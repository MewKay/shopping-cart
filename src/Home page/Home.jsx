import { Link } from "react-router-dom";
import styles from "./Home.module.css";

const Home = function createHomeComponent() {
  return (
    <main className={styles["main-container"]}>
      <h2 className={styles["hero-head"]}>Welcome to our store!</h2>
      <p className={styles["hero-intro"]}>
        Lorem ipsum, dolor sit amet consectetur adipisicing elit. Ducimus autem
        itaque corporis vitae id vero quaerat possimus nisi neque alias?
      </p>
      <Link to={"store"} className={styles["store-link"]}>
        Go to store
      </Link>
    </main>
  );
};

export default Home;
