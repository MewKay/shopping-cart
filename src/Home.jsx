const Home = function createHomeComponent() {
  return (
    <>
      <h2>Welcome to our store!</h2>
      <p>
        Lorem ipsum, dolor sit amet consectetur adipisicing elit. Ducimus autem
        itaque corporis vitae id vero quaerat possimus nisi neque alias?
      </p>
      <button>Go to store</button>
    </>
  );
};

export default Home;
