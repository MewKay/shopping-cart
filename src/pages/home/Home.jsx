import { Link } from "react-router-dom";
import styles from "./Home.module.css";
import { ArrowRight } from "lucide-react";

const Home = function createHomeComponent() {
  return (
    <main className={styles["main-container"]}>
      <h2 className={styles["hero-head"]}>Welcome to our store!</h2>
      <p className={styles["hero-intro"]}>
        Lorem ipsum, dolor sit amet consectetur adipisicing elit. Ducimus autem
        itaque corporis vitae id vero quaerat possimus nisi neque alias?
      </p>
      <Link to={"store"} className={styles["store-link"]}>
        <p className={styles["link-text"]}>Go to store</p>
        <ArrowRight size={28} className={styles["link-icon"]} />
      </Link>
    </main>
  );
};

export default Home;
