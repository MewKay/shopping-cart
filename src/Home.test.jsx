import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import Home from "./Home";

describe("Home component", () => {
  it("renders welcome title, store description and going to store button", () => {
    const { container } = render(<Home />);

    expect(container).toMatchSnapshot();
  });
});
