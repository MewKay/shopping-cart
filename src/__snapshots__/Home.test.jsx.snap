// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home component > renders welcome title, store description and going to store button 1`] = `
<div>
  <h2>
    Welcome to our store!
  </h2>
  <p>
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Ducimus autem itaque corporis vitae id vero quaerat possimus nisi neque alias?
  </p>
  <button>
    Go to store
  </button>
</div>
`;
